PMM-5052 documentation about textfile collector usage in pmm2 (#601)
percona/pmm-doc: docs/how-to/extend-metrics.md

# Extend Metrics

Sometimes, though, you’ll have the need for a metric that’s not present in the list of node_exporter metrics out of the box. 
The textfile collector allows exporting of statistics from batch jobs. It can also be used to export static metrics, such as what role a machine has. 

## Enable the textfile collector

The collector is enabled by default. Next folders are used for different resolutions:

| Resolution | Folder                                                                  |
|------------|-------------------------------------------------------------------------|
|  High      | /usr/local/percona/pmm2/collectors/textfile-collector/high-resolution   |
|  Medium    | /usr/local/percona/pmm2/collectors/textfile-collector/medium-resolution |
|  Low       | /usr/local/percona/pmm2/collectors/textfile-collector/low-resolution    |

![!image](../_images/node-exporter.textfile-collector.1.png)

Exporter parse all files in that directories matching the glob *.prom using the text format.

> 

Reviewer: ```suggestion
Files in the folders are expected to be parsed in [Exposition format](https://prometheus.io/docs/instrumenting/exposition_formats/) and metrics stored on PMM Server-side with an additional label related to this Node.

```

Author: I have added the link and the last sentence.

![Screenshot 2021-09-29 at 11 56 42](https://user-images.githubusercontent.com/25184101/135237014-ab10e53e-d0f4-4dd7-b41c-dee32e33b1b3.png)

# Extend Metrics

> Sometimes, though, you’ll have the need for a metric that’s not present in the list of node_exporter metrics out of the box. 

Reviewer 2: ```suggestion
When you need a metric that’s not present in the default list of `node_exporter` metrics you may be able to use the `textfile` collector. 
```

The textfile collector allows exporting of statistics from batch jobs. It can also be used to export static metrics, such as what role a machine has. 

## Enable the textfile collector

> The collector is enabled by default. Next folders are used for different resolutions:

Reviewer 2: ```suggestion
The collector is enabled by default. The following folders are used for different resolutions:
```

| Resolution | Folder                                                                  |
|------------|-------------------------------------------------------------------------|
|  High      | /usr/local/percona/pmm2/collectors/textfile-collector/high-resolution   |
|  Medium    | /usr/local/percona/pmm2/collectors/textfile-collector/medium-resolution |
|  Low       | /usr/local/percona/pmm2/collectors/textfile-collector/low-resolution    |

![!image](../_images/node-exporter.textfile-collector.1.png)

> Exporter parse all files in that directories matching the glob *.prom using a simple text-based [exposition format](https://prometheus.io/docs/instrumenting/exposition_formats/#text-based-format).

Reviewer 2: ```suggestion
The exporter parses all files in these directories that match the filename wildcard expression `*.prom` using a simple text-based [exposition format](https://prometheus.io/docs/instrumenting/exposition_formats/#text-based-format).
```

> Metrics are stored on PMM Server-side with additional labels related to this Node.

Reviewer 2: ```suggestion
Metrics are stored on the PMM Server-side with additional labels related to this Node.
```

## Examples of shell commands for custom metrics

To statically set roles for a machine using labels:

>     ```sh

Reviewer 2: ```suggestion
```sh
```

echo 'node_role{role="my_monitored_server_1"} 1' > /usr/local/percona/pmm2/collectors/textfile-collector/low-resolution/node_role.prom
```

> To atomically push loggedin users for a cron job:

Reviewer 2: ```suggestion
Here's an example of a cron job that atomically pushes logged-in users:
```

    ```sh
    $ cat /etc/cron.d/loggedin_users

>     */1 * * * *     root    /usr/bin/who | /usr/bin/wc -l | sed -ne 's/^/node_loggedin_users /p' > /usr/local/percona/pmm2/collectors/textfile-collector/high-resolution/node_users.prom

Reviewer 2: This command doesn't look right to me. The output from `cat /etc/cron.d/loggedin_users` is what? Then the part beginning `/usr/bin/who ...` comes where?

Reviewer 2: When fixed, this code block doesn't need to be indented.

Author: It's an example of a cron job. I can remove the "cat" part but I think people would be more confused. 

Let's add the next screenshot.
![Screenshot 2021-09-29 at 12 23 06](https://user-images.githubusercontent.com/25184101/135241909-01a12517-693e-4d56-b237-e2d1e8c0f5eb.png)

Reviewer 2: I see, the whole pipeline is one big cron job...
